rotkehlchen/chain/ethereum/modules/compound/decoder.py

```diff
@@ -252,13 +252,12 @@ def decode_comp_claim(
         # Transactions with comp claim have many such "distributed" events. We need to do a
         # decoded evens iteration only at the end but can't think of a good way to avoid
         # the possibility of checking all such events
-
         supplier_address = hex_or_bytes_to_address(tx_log.topics[2])
         if not self.base.is_tracked(supplier_address):
             return None, []
 
         for event in decoded_events:
-            if event.event_type == HistoryEventType.RECEIVE and event.location_label == supplier_address and event.asset == A_COMP and event.counterparty == COMPTROLLER_PROXY_ADDRESS:  # noqa: E501
+            if event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and event.location_label == supplier_address and event.asset == A_COMP and event.address == COMPTROLLER_PROXY_ADDRESS:  # noqa: E501
```

Comment:
why adding this subtype check here?

Reply:
That is the neat part.

There is a `counterparty` check. Previously if there was a match, counterparty (which is initially an address) was overwritten by a protocol label and thus counterparty was not matched in the next calls to this function.

Now, since we don't modify the counterparty anymore, for a particular event this condition will always succeed. And to prevent that I added a subtype check to check whether an event was decoded or not.